components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/IdentityProviderManager.java

```diff
@@ -1673,11 +1673,18 @@ public IdentityProvider getIdPByRealmId(String realmId, String tenantDomain)
     public IdentityProvider getEnabledIdPByRealmId(String realmId, String tenantDomain)
             throws IdentityProviderManagementException {
 
-        IdentityProvider idp = getIdPByRealmId(realmId, tenantDomain);
-        if (idp != null && idp.isEnable()) {
-            return idp;
+        int tenantId = IdentityTenantUtil.getTenantId(tenantDomain);
+        if (StringUtils.isEmpty(realmId)) {
+            String msg = "Invalid argument: Identity Provider Home Realm Identifier value is empty";
+            throw new IdentityProviderManagementException(msg);
```

Comment:
Pass the message directly to the error, no need to create an variable

Reply:
Fixed with [50f95c5](https://github.com/wso2/carbon-identity-framework/pull/4991/commits/50f95c50707223620d969230836a74dad28ded0f)